plugins/registry.js

```diff
@@ -778,7 +778,18 @@ export default class PluginRegistry {
     // - showTeamSidebar - A flag to display or hide the team sidebar in products. Defaults to false (hidden).
     // All parameters are required.
     // Returns a unique identifier.
-    registerProduct(baseURL, switcherIcon, switcherText, switcherLinkURL, mainComponent, headerCentreComponent = () => null, headerRightComponent = () => null, showTeamSidebar = false) {
+    registerProduct(
```

Comment:
I would love to see that, too, though there's a lot of methods with many args. Perhaps in separate effort we could "object-ize" a parallel PluginRegistry and soft deprecate this one.

Reply:
Either that or I was thinking of making a helper that converts a potential array of args into an object. That's something I've wanted in the client package as well since some of those API calls are similarly getting massive. Maybe that'd be a good distraction one day since I don't think it'd be super difficult to do